ui-test/public-suite/extension-activity-test.ts

```diff
@@ -0,0 +1,59 @@
+/*-----------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *-----------------------------------------------------------------------------------------------*/
+import { SideBarView, ActivityBar, WebDriver, VSBrowser, EditorView } from 'vscode-extension-tester';
+import { expect } from 'chai';
+import { viewHasItems } from '../common/conditions';
+import { views } from '../common/constants';
+
+export function extensionActivityTest() : void{
+
+  describe('Tekton Activity Test', () => {
+    let driver: WebDriver;
+
+    before(async function() {
+      this.timeout(200000);
+      driver = VSBrowser.instance.driver;
+      await new EditorView().closeAllEditors();
+    });
+
+    it('Check Tekton Pipelines Exists', async function(){
+      this.timeout(300000);
+      const tektonPip = await new ActivityBar().getViewControl(views.TEKTON_TITLE);
+      expect(tektonPip).not.undefined;
+    });
+
+    it('Check Tekton View Categories', async function(){
+      this.timeout(300000);
+      await (await new ActivityBar().getViewControl(views.TEKTON_TITLE)).openView();
+      await driver.wait(() => { return viewHasItems(); }, 200000);
+      const tektonCats = await new SideBarView().getContent().getSections();
+      expect(tektonCats.length).equals(3);
+
+      for (let i = 0; i < tektonCats.length; i++){
```

Comment:
Not  really resilient test case. If the order changes it will start failing. I would suggest to use map to get all items from the object (categories) and to.have.members chai assertion (https://www.chaijs.com/api/bdd/). this way you can check for multiple elements in array.